repo/datastore.go

```diff
@@ -49,11 +48,29 @@ type ThreadStore interface {
 	DeleteByName(name string) error
 }
 
+type DeviceStore interface {
+	Queryable
+	Add(device *Device) error
+	Get(id string) *Device
+	GetByName(name string) *Device
+	List(query string) []Device
+	Delete(id string) error
+	DeleteByName(name string) error
+}
+
+type PeerStore interface {
```

Comment:
`PeerStore` tries to keep track of who is in a thread because they each need to be contacted directly and sent updates. If this becomes cumbersome for large threads, we could institute some rules like:
- only send to 8 random peers
- if you get an update that is new to you, relay it to 8 random peers
This is sort of a cheap DHT... similar to what gundb does

Reply:
That said, I don't think we'll need to deal w/ that for a good while